arm-powerbiembedded/2016-01-29/swagger/powerbiembedded.json

```diff
@@ -0,0 +1,1127 @@
+                "operationId": "checkNameAvailability",
+                "description": "The **Check Power BI Workspace Collection Name Availability** operation checks that the specified Power BI Workspace Collection name is valid and not in use.",
+                "consumes": [
+                    "application/json"
+                ],
+                "produces": [
+                    "application/json"
+                ],
+                "parameters": [
+                    {
+                        "name": "subscriptionId",
+                        "in": "path",
+                        "description": "Azure subscription id",
+                        "required": true,
+                        "type": "string"
+                    },
+                    {
+                        "name": "location",
+                        "in": "path",
+                        "description": "Azure location",
+                        "required": true,
+                        "type": "string"
+                    },
+                    {
+                        "name": "api-version",
+                        "in": "query",
+                        "description": "Azure API version",
+                        "required": true,
+                        "type": "string"
+                    },
+                    {
+                        "name": "body",
+                        "in": "body",
+                        "description": "Check name availability request",
+                        "required": true,
+                        "schema": {
+                            "$ref": "#/definitions/CheckNameRequest"
+                        }
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "Request completed successfully",
+                        "schema": {
+                            "$ref": "#/definitions/CheckNameResponse"
+                        }
+                    },
+                    "400": {
+                        "description": "Request is not well formed"
+                    },
+                    "403": {
+                        "description": "Authentication failed"
+                    },
+                    "404": {
+                        "description": "Subscription does not exist"
+                    },
+                    "408": {
+                        "description": "Operation could not be completed within the permitted time"
+                    },
+                    "503": {
+                        "description": "Service unavailable"
+                    }
+                }
+            }
+        },
+        "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.PowerBI/workspaceCollections": {
+            "get": {
+                "tags": [
+                    "WorkspaceCollections"
+                ],
+                "operationId": "getWorkspacesCollectionsInResourceGroup",
```

Comment:
@wbreza @mattmazzola- For Azure services we follow the convention of noun_verb for operationId.
So the operastionId should be:
WorkspaceCollections_listByResourceGroup
This will translate to client.WorkspaceCollections.ListByResourceGroup(). This provides better intellisense experience rather than all the methods being on the client.

All of this is very well documented. Please read this https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/creating-swagger.md#operation-and-operationid